dpctl/apis/include/dpctl4pybind11.hpp

```diff
@@ -671,6 +671,54 @@ struct type_caster<sycl::kernel_bundle<sycl::bundle_state::executable>>
     DPCTL_TYPE_CASTER(sycl::kernel_bundle<sycl::bundle_state::executable>,
                       _("dpctl.program.SyclProgram"));
 };
+
+/* This type caster associates
+ * ``sycl::half`` C++ class with Python :class:`float` for the purposes
+ * of generation of Python bindings by pybind11.
+ */
+template <> struct type_caster<sycl::half>
+{
+public:
+    bool load(handle src, bool convert)
+    {
+        double py_value;
+
+        if (!src) {
+            return false;
+        }
+
+        PyObject *source = src.ptr();
+
+        if (convert || PyFloat_Check(source)) {
+            py_value = PyFloat_AsDouble(source);
+        }
+        else {
+            return false;
+        }
+
+        bool py_err = (py_value == double(-1)) && PyErr_Occurred();
+
+        if (py_err) {
+            PyErr_Clear();
+            if (py_err && convert && (PyNumber_Check(source) != 0)) {
```

Comment:
```suggestion
            if (convert && (PyNumber_Check(source) != 0)) {
```

We checked thar ``py_err`` has value of `true`.